cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/index.tsx

```diff
@@ -78,66 +79,50 @@ const WidgetCollectionView: React.FC<IWidgetCollectionProps> = ({
   widgetToAttributes,
   setWidgetToAttributes,
 }) => {
-  const [activeWidgets, setActiveWidgets] = React.useState(groupID ? groupToWidgets[groupID] : []);
+  const activeWidgets = groupID ? groupToWidgets.get(groupID) : fromJS([]);
   const [openWidgetIndex, setOpenWidgetIndex] = React.useState(null);
 
-  React.useEffect(() => {
-    if (groupID) {
-      setActiveWidgets(groupToWidgets[groupID]);
-    } else {
-      setActiveWidgets([]);
-    }
-  }, [groupToWidgets]);
-
   function addWidgetToGroup(index: number) {
     return () => {
       const newWidgetID = 'Widget_' + uuidV4();
 
       // Add a new widget's ID at the specified index
-      const newWidgets = [...activeWidgets];
-      if (newWidgets.length === 0) {
-        newWidgets.splice(0, 0, newWidgetID);
+      let newWidgets;
+      if (activeWidgets.size === 0) {
```

Comment:
you can also use isEmpty() to check for an empty Map or List

Reply:
Resolved!